### DISCUSSION TITLE
Allow duplicate options with the same value / Preserved original select options

### DISCUSSION THREAD

**Discussion by nullifye** [orchidjs/tom-select#395] [Q&A]
How to allow duplicate options with the same value. my code is like this:

```
<div class="p-4"><select id="ex" autocomplete="off" multiple>
      <option value="c1">Cool1</option>
      <option value="c1">Cool2</option>
      <option value="c3">Cool3</option>
</select></div>
```

Right now with default configuration, **Cool2** option will be omitted/removed by tom-select since it has the same value with Cool1. I need to preserved the original options.

**Comment by iamB0rgy** [comment]
Add this parameter, `duplicates: true,`

**Reply by abdullzz** [reply]
no longer works i guess

**Reply by hawkeye116477** [reply]
Looks like it only allows selecting the same option more than once, but that option will be displayed multiple times under same name, but we need different names with same values.

**Comment by interceptor** [comment]
Damn, that's not handy for this scenario:

function addOptions(tomSelectElement) {
    const options = [
        {value: 1, text: "1 hour"},
        {value: 3, text: "3 hours"},
        {value: 7, text: "6 hours"},
        {value: 10, text: "12 hours"},
        {value: 1, text: "1 day"},
    ];
    options.forEach(option => tomSelectElement.addOption(option));
}

**Comment by GibzClt** [comment]
@oyejorge Do you have a suggestion how we can this to work with tom-select ?

**Comment by cuu508** [comment]
Option ids must be unique. If you add an option with the same id as an existing option, Tom Select will update the existing option instead of adding a duplicate.

There's a workaround when using option groups: the same option can appear in multiple option groups.

Let's say you have a selector of animals and you want "Pegasus" to appear under both "mammal" and "bird" categories. The following **would not** work – the "Pegasus" option would appear in the dropdown only once:

```html
<!DOCTYPE html>
<html>
<head>
    <link href="https://cdn.jsdelivr.net/npm/tom-select/dist/css/tom-select.default.css" rel="stylesheet">
</head>
<body>
    <select></select>
    <script src="https://cdn.jsdelivr.net/npm/tom-select/dist/js/tom-select.complete.js"></script>
    <script>
        new TomSelect('select',{
            options: [
                {class: 'mammal', value: "pegasus", name: "Pegasus"},
                {class: 'mammal', value: "cat", name: "Cat" },
                {class: 'bird', value: "pegasus", name: "Pegasus"},
                {class: 'bird', value: 'duck', name: 'Duck'},
            ],
            optgroups: [
                {value: 'mammal', label: 'Mammal'},
                {value: 'bird', label: 'Bird'},
            ],
            optgroupField: 'class',
            labelField: 'name',
            searchField: ['name'],
        });
    </script>
</body>
</html>
```

What you can do instead is define the "Pegasus" option only once, but assign multiple categories to it, using an array:

```html
<!DOCTYPE html>
<html>
<head>
    <link href="https://cdn.jsdelivr.net/npm/tom-select/dist/css/tom-select.default.css" rel="stylesheet">
</head>
<body>
    <select></select>
    <script src="https://cdn.jsdelivr.net/npm/tom-select/dist/js/tom-select.complete.js"></script>
    <script>
        new TomSelect('select',{
            options: [
                {class: ['mammal', 'bird'], value: "pegasus", name: "Pegasus" },
                {class: 'mammal', value: "cat", name: "Cat" },
                {class: 'bird', value: 'duck', name: 'Duck'},
            ],
            optgroups: [
                {value: 'mammal', label: 'Mammal'},
                {value: 'bird', label: 'Bird'},
            ],
            optgroupField: 'class',
            labelField: 'name',
            searchField: ['name'],
        });
    </script>
</body>
</html>
```
